internal/analytics/history.go

```
@@ -0,0 +1,271 @@
+
+// snapshotRowCounts opens a snapshot read-only and returns per-table row counts
+// for all tables found in sqlite_master.
+func snapshotRowCounts(absPath string) (map[string]int64, error) {
+	dsn := "file:" + absPath + "?mode=ro&_pragma=query_only(true)"
+	db, err := sql.Open("sqlite", dsn)
+	if err != nil {
+		return nil, fmt.Errorf("open %s: %w", absPath, err)
+	}
+	defer db.Close()
+
+	if err := db.Ping(); err != nil {
```

golangci-lint (2.12.2)

[error] (*database/sql.DB).Ping must not be called. use (*database/sql.DB).PingContext

(noctx)